### PR TITLE
Create Pollard's_rho_algorithm.py

### PR DESCRIPTION
**Pull request by rupal121**
issue #243 
Implemented Pollard's rho algorithm

### CODE REVIEW

**Comment by rupal121** [comment]
what changes you want? @rcoder23

**Comment by rcoder23** [comment]
@rupal121 Use docstring to explain code .

**Comment by rupal121** [comment]
@rcoder23 Kindly review

File: Algorithms/Integer Factorization Algorithm/Pollard's_rho_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        d ← gcd(|x - y|, n)   # gcd of absolute difference of x & y and of num is taken
+
+    if d = n:       #if gcd is equal to n then there no factor exist other one and itself
+        return failure
+    else:
+        return d
+'''
+#import needed library
+from itertools import count
+from math import gcd
```
Comment (rohanbabbar04) [inline Algorithms/Integer Factorization Algorithm/Pollard's_rho_algorithm.py:21]: Use from fractions import gcd

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]
Add a sample output at the end of the code
Use Dynamic output

**Comment by rupal121** [comment]
All the changes have been made. Kindly review @rcoder23 @rohanbabbar04

**Review by rcoder23 — CHANGES_REQUESTED** [review]
define what is Pollard's rho algorithm and add it in algorithms folder .without sub folder

**Comment by rupal121** [comment]
All suggested changes have been made. Kindly review @rcoder23 @rohanbabbar04

**Review by rohanbabbar04 — APPROVED** [review]

**Comment by rupal121** [comment]
Please list all the changes you want this time @rcoder23  so that I can finalize this issue after this.

File: Algorithms/Pollard's_rho_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+for cycle in count(1):
+#here we assign y=x
+    y = x
+    for i in range(2 ** cycle):
+# we use polynomial function to find factor 
+# function, y=(x*x+c)%number
+        x = (x * x + 1) % number
+#take gcd of absolute difference of x&y and number
+#gcd is treated as a factor if it is not unity
+        factor = gcd(x - y, number)
```
Comment (rcoder23) [inline Algorithms/Pollard's_rho_algorithm.py:42]: use Math.gcd

File: Algorithms/Pollard's_rho_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    else:
+        return d
+'''
+#import needed library
+from itertools import count
+from fractions import gcd
+import sys
+#take input "number" whose factor we have to determine
+#take any value for x and c, here c=1 is taken
+number=int(input())
```
Comment (rcoder23) [inline Algorithms/Pollard's_rho_algorithm.py:30]: Take input with message

**Review by rcoder23 — CHANGES_REQUESTED** [review]

**Comment by rupal121** [comment]
All changes have been made except "use math.gcd"  because @rohanbabbar04 said to use fractions library and you are asking to use math.gcd @rcoder23. Now please approve it @rcoder23.

**Review by rcoder23 — APPROVED** [review]

**Pull request merged**